library/src/main/java/com/schibsted/spain/barista/internal/matcher/TextColorMatcher.kt

```diff
@@ -45,8 +45,15 @@ class TextColorMatcher(private val expectedColor: Int) : BoundedMatcher<View, Te
 
   private fun matchesColorList(textView: TextView): Boolean {
     val currentColorList = textView.textColors
-    val expectedColorInt = ContextCompat.getColorStateList(textView.context, expectedColor)
-    return currentColorList == expectedColorInt
+    val expectedColorList = ContextCompat.getColorStateList(textView.context, expectedColor)!!
+
+    val allStates = ALL_COLOR_STATE_LIST_STATES.flatMap { state -> listOf(state, -state) }.map { state -> intArrayOf(state) }
```

Comment:
What does this `-` of `listOf(state, -state)` mean? Just curiosity

Reply:
It reverses the state:

state enabled: `android.R.attr.state_enabled`
state disabled: `-android.R.attr.state_enabled`

Reply:
WOW, that's a big WTF. What's the magic behind that?

IMHO we could do something easier to catch...

Reply:
What do you mean?

Reply:
https://stackoverflow.com/a/5368765/1798470